GlucoseOnFhir/GlucoseMeterViewController.swift

```diff
@@ -86,42 +94,92 @@ class GlucoseMeterViewController: UITableViewController, GlucoseProtocol, FHIRPr
     
     func glucoseFeatures(features:GlucoseFeatures) {
         glucoseFeatures = features
-        
-        self.refreshTable()
     }
     
     public func glucoseMeterDidTransferMeasurements(error: NSError?) {
-        SwiftSpinner.show(duration: 1.5, title: "Download Complete", animated: false)
-        SwiftSpinner.show("Connecting to FHIR", animated: true)
+        print("glucoseMeterDidTransferMeasurements")
         
-        self.searchForPatient()
+        DispatchQueue.once(executeToken: "glucoseOnFhir.glucoseMeterDidTransferMeasurements.runOnce") {
+            self.searchForPatient(given: self.givenName, family: self.familyName) { (bundle, error) -> Void in
+                if(bundle?.entry != nil) {
+                    self.searchForObservations()
+                }
+            }
+            self.searchForDevice() { (bundle, error) -> Void in }
+        }
+    }
+    
+    func searchForObservations() {
+        for measurement in self.glucoseMeasurements {
+            self.searchForObservation(measurement: measurement) { (bundle,error) -> Void in
+                if (bundle?.entry == nil) {
+                    print("measurement \(measurement.sequenceNumber) not found")
+                    self.logEvent(event: "measurement \(measurement.sequenceNumber) not found")
+                    measurement.existsOnFHIR = false
+                    self.refreshTable()
+                } else {
+                    print("measurement \(measurement.sequenceNumber) found")
+                    self.logEvent(event: "measurement \(measurement.sequenceNumber) found")
+                    measurement.existsOnFHIR = true
+                    measurement.fhirID = bundle?.entry?.first?.resource?.id
+                    
+                    if(bundle?.entry != nil) {
+                        let observations = bundle?.entry?
+                            .filter() { return $0.resource is Observation }
+                            .map() { return $0.resource as! Observation }
+                        
+                        self.observations.append((observations?.first)!)
+                        print("observations count: \(self.observations.count)")
+                    }
+                    self.refreshTable()
+                }
+            }
+        }
     }
     
     public func glucoseError(error: NSError) {
         print("glucoseError")
     }
     
+    func createActivityView() -> UIActivityIndicatorView {
+        let actInd: UIActivityIndicatorView = UIActivityIndicatorView()
```

Comment:
use `let activity = UIActivityIndicatorView(frame: .zero)` and `activity.sizeToFit()`